### PR TITLE
Add Preset Management UI

### LINKED ISSUE #2798: [UI] Presets Management in the Admin Panel
**What would you like to be added? / User Story**:
It should be possible to manage presets via the UI in the admin panel.

#### Limitations
For security reasons when editing a preset, it will not be possible to see existing preset information. It will only be possible to override it by providing a brand new set of information.

#### Note
The preset object contains a set of supported providers. It will not be possible to only create a preset without providing at least 1 provider information. It means that adding another provider to the preset only updates the preset object. It does not create a new preset.

#### Acceptance Criteria
- [x] List presets for all providers
  - [ ] ~Filter by enabled status~
- [x] List presets for selected provider
  - [x] Filter by datacenter
  - [ ] ~Filter by enabled status~
- [x] Create a preset for the provider
  - [x] Restrict preset to selected datacenter
  - [x] Mark created preset as enabled/disabled
- [x] Update (override) existing provider preset information
  - [x] Enable/disable preset
  - [x] Update/remove datacenter restriction

### PR DESCRIPTION
**Pull request by floreks**
**What this PR does / why we need it**:
- [x] List presets for all providers
- [x] List presets for the selected provider
  - [x] Filter by datacenter
- [x] Create a preset for the provider
  - [x] Restrict preset to selected datacenter
  - [x] Mark created preset as enabled/disabled
- [x] Update (override) existing provider preset information
  - [x] Enable/disable preset
  - [x] Update/remove datacenter restriction

**Which issue(s) this PR fixes**:
Fixes #2798

**Release note**:

```release-note
Added Preset management UI to the admin settings.
```

### CODE REVIEW

File: src/app/settings/admin/presets/theme.scss [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+@mixin theme-preset-list-component($colors) {
+  .mat-card-content {
+    button {
+      color: inherit;
```
Comment (maciaszczykm) [inline src/app/settings/admin/presets/theme.scss:15]: This could be moved to the component style I think as there is no color value used.

**Review by maciaszczykm — COMMENTED** [review]
A lot of work done here 👍🏻 Codewise I have only one remark, I will do some more manual tests.

File: src/app/settings/admin/presets/theme.scss (hunk 1 above)
Comment (floreks) [reply src/app/settings/admin/presets/theme.scss:15]: It was done intentionally as I try not to mix `color`, `background-color`, or any other property that is related to theming with regular style files.

**Pull request merged**